### ISSUE TITLE
Composer 2.0 compatibility

### ISSUE THREAD

**Issue by cebe** [yiisoft/yii2-composer#27]
Composer 2.0 is going to be released soon. Here is the upgrade guide:

https://github.com/composer/composer/blob/master/UPGRADE-2.0.md#upgrades-guides-for-composer-1x-to-20

> composer-plugin-api has been bumped to 2.0.0

we currently require `^1.0.0`

https://github.com/yiisoft/yii2-composer/blob/99e52c7aa9130ed14a53953696a1542d9978096f/composer.json#L25

> PluginInterface has new methods

need to implement these.

Possibly more things to do, need to check full upgrade guide.

**Comment by cebe** [comment]
Probably worth a 2.1.0 release.

**Comment by schmunk42** [comment]
I haven't tried composer 2.0 yet, but I think it's gonna be problematic with npm packages, due to this issue: https://github.com/fxpio/composer-asset-plugin/issues/333

```
Deprecation warning: require.npm-asset/fullcalendar--core is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
```

Yii core does not use npm packages AFAIK, but packages from the ecosystem which use npm-assets won't work, I think. See also https://github.com/hiqdev/asset-packagist/issues/103#issuecomment-555049342

**Comment by cebe** [comment]
@schmunk42 I made an issue on composer repo: https://github.com/composer/composer/issues/8749
